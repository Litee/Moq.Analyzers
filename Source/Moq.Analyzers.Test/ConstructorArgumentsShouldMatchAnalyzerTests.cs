namespace Moq.Analyzers.Test
{
    using System.IO;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using Microsoft.CodeAnalysis.Diagnostics;
    using TestHelper;
    using Xunit;

    [UseReporter(typeof(DiffReporter))]
    public class ConstructorArgumentsShouldMatchAnalyzerTests : DiagnosticVerifier
    {
        [Fact]
        public void ShouldFailIfClassParametersDoNotMatch()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/ConstructorArgumentsShouldMatch.cs")));
        }

        [Fact]
        public void ShouldPassIfCustomMockClassIsUsed()
        {
           // Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockInterfaceWithParametersCustomMockFile.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new ConstructorArgumentsShouldMatchAnalyzer();
        }
    }
}