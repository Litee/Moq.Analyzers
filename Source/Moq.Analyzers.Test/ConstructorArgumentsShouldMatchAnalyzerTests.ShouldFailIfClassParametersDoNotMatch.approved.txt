Diagnostic 1
	Id: Moq1000
	Location: SourceFile(Test0.cs[928..937))
	Highlight: (1, true)
	Lines: var mock1 = new Moq.Mock<Foo>(1, true);
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 2
	Id: Moq1000
	Location: SourceFile(Test0.cs[1009..1018))
	Highlight: (2, true)
	Lines: var mock2 = new Mock<ConstructorArgumentsShouldMatch.Foo>(2, true);
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 3
	Id: Moq1000
	Location: SourceFile(Test0.cs[1090..1098))
	Highlight: ("1", 3)
	Lines: var mock3 = new Mock<ConstructorArgumentsShouldMatch.Foo>("1", 3);
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 4
	Id: Moq1000
	Location: SourceFile(Test0.cs[1170..1193))
	Highlight: (new int[] { 1, 2, 3 })
	Lines: var mock4 = new Mock<ConstructorArgumentsShouldMatch.Foo>(new int[] { 1, 2, 3 });
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 5
	Id: Moq1000
	Location: SourceFile(Test0.cs[1290..1320))
	Highlight: (MockBehavior.Strict, 4, true)
	Lines: var mock1 = new Mock<Foo>(MockBehavior.Strict, 4, true);
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 6
	Id: Moq1000
	Location: SourceFile(Test0.cs[1396..1425))
	Highlight: (MockBehavior.Loose, 5, true)
	Lines: var mock2 = new Moq.Mock<ConstructorArgumentsShouldMatch.Foo>(MockBehavior.Loose, 5, true);
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 7
	Id: Moq1000
	Location: SourceFile(Test0.cs[1501..1529))
	Highlight: (MockBehavior.Loose, "2", 6)
	Lines: var mock3 = new Moq.Mock<ConstructorArgumentsShouldMatch.Foo>(MockBehavior.Loose, "2", 6);
	Severity: Warning
	Message: Sealed classes cannot be mocked.

